### PR TITLE
Optimize track menu.

### PR DESCRIPTION
**Pull request by dyphire**
Inspired by https://github.com/stax76/mpv.net/pull/431#issuecomment-1144537270
Add encoding information to subtitles tracks.
Add title for external tracks, match and replace the filename to avoid too long characters.

### CODE REVIEW

File: src/Misc/Player.cs [hunk 1]
```diff
@@ -1407,6 +1411,7 @@ public List<MediaTrack> GetExternalTracks()
                         Add(track, GetPropertyString($"track-list/{i}/codec").ToUpperEx());
                         Add(track, GetPropertyInt($"track-list/{i}/audio-channels") + " channels");
                         Add(track, "External");
+                        Add(track, $"{title}");
```
Comment (Evengard) [inline src/Misc/Player.cs:1414]: Why not just `Add(track, title);` ?

File: src/Misc/Player.cs [hunk 2]
```diff
@@ -1416,7 +1421,9 @@ public List<MediaTrack> GetExternalTracks()
                     {
                         MediaTrack track = new MediaTrack();
                         Add(track, GetLanguage(GetPropertyString($"track-list/{i}/lang")));
+                        Add(track, GetPropertyString($"track-list/{i}/codec").ToUpperEx());
                         Add(track, "External");
+                        Add(track, $"{title}");
```
Comment (Evengard) [inline src/Misc/Player.cs:1426]: Same here...

**Comment by stax76** [comment]
I've rewritten much code now however because of a new feature:

- New `media-info` option allowing to fully disable the usage of the
  media info library, media info is disabled by default and considered
  to be deprecated.

Hopefully it's not buggy.

**Comment by dyphire** [comment]
> * New `media-info` option allowing to fully disable the usage of the
>   media info library, media info is disabled by default and considered
>   to be deprecated.

    
Relatively speaking, I personally prefer the track-menu information displayed by MediaInfo. 
It's video and audio tracks information is richer than mpv track-list.
In particular, the 'track-list/N/demux-bitrate' of the track-list is almost unavailable.I think this is an important audio content information.

**Comment by dyphire** [comment]
In addition, the '__' should replace back '_' when the new `show-media-info` is displayed.

**Comment by dyphire** [comment]
There is a new bug. When media info is enabled, show media info will not display external track information. 
If try again after opening the show-menu, the external track information will be displayed.

**Comment by Evengard** [comment]
Probably the `_` => `__` filtering should be done directly in the wpf element assignement...

**Comment by stax76** [comment]
> Probably the _ => __ filtering should be done directly in the wpf element assignement...

Yes, it does this now.

**Pull request merged**